testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/EEConcurrencyExecutorShutdownTestCase.java

```diff
@@ -0,0 +1,211 @@
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.TIMEOUT;
+import static org.jboss.as.controller.operations.common.Util.createAddOperation;
+import static org.jboss.as.controller.operations.common.Util.createEmptyOperation;
+import static org.jboss.as.test.integration.domain.management.util.DomainTestSupport.cleanFile;
+import static org.jboss.as.test.integration.domain.management.util.DomainTestUtils.executeForResult;
+import static org.junit.Assert.fail;
+
+/**
+ * The test case schedules an instance of task ReSchedulingTask.
+ * Each instance of ReSchedulingTask sleeps for 10 seconds and then re-schedules another instance of its own class.
+ * After the CLI command /host=master/server-config=server-one:stop() is invoked the server should stop.
+ * Test for [ WFCORE-3868 ].
+ *
+ * @author Daniel Cihak
+ */
+public class EEConcurrencyExecutorShutdownTestCase {
+
+    private static final String ARCHIVE_FILE_NAME = "test.war";
+    private static final PathAddress ROOT_DEPLOYMENT_ADDRESS = PathAddress.pathAddress(DEPLOYMENT, ARCHIVE_FILE_NAME);
+    private static final PathAddress MAIN_SERVER_GROUP_ADDRESS = PathAddress.pathAddress(SERVER_GROUP, "main-server-group");
+    private static final PathAddress MAIN_SERVER_GROUP_DEPLOYMENT_ADDRESS = MAIN_SERVER_GROUP_ADDRESS.append(DEPLOYMENT, ARCHIVE_FILE_NAME);
+    public static final String FIRST_SERVER_NAME = "main-one";
+
+    private static DomainTestSupport testSupport;
+    public static DomainLifecycleUtil domainMasterLifecycleUtil;
+    private static File tmpDir;
+
+    private static DomainClient masterClient;
+
+    @BeforeClass
+    public static void setupDomain() {
+        testSupport = createAndStartDefaultEESupport(EEConcurrencyExecutorShutdownTestCase.class.getSimpleName());
+        domainMasterLifecycleUtil = testSupport.getDomainMasterLifecycleUtil();
+        masterClient = domainMasterLifecycleUtil.getDomainClient();
+    }
+
+    private static DomainTestSupport createAndStartDefaultEESupport(final String testName) {
+        try {
+            final DomainTestSupport.Configuration configuration;
+            if (Boolean.getBoolean("wildfly.master.debug")) {
+                configuration = DomainTestSupport.Configuration.createDebugMaster(testName,
+                        "domain-configs/domain-standard-ee.xml", "host-configs/host-master.xml", "host-configs/host-slave.xml");
+            } else if (Boolean.getBoolean("wildfly.slave.debug")) {
+                configuration = DomainTestSupport.Configuration.createDebugSlave(testName,
+                        "domain-configs/domain-standard-ee.xml", "host-configs/host-master.xml", "host-configs/host-slave.xml");
+            } else {
+                configuration = DomainTestSupport.Configuration.create(testName,
+                        "domain-configs/domain-standard-ee.xml", "host-configs/host-master.xml", "host-configs/host-slave.xml");
```

Comment:
If you agree, would you mind using `null` as a value for `slaveConfig` argument on those three configurations? Really we don't need to start a slave for this test, doing so we can save some seconds

Reply:
@yersan Yes, no problem. I have changed the configurations in the PR.